adm_program/modules/dates/dates_function.php

```diff
@@ -89,9 +89,21 @@
     $_SESSION['dates_request'] = $_POST;
 
     // ------------------------------------------------
-    // pruefen ob alle notwendigen Felder gefuellt sind
+    // check if all necessary fields are filled
     // ------------------------------------------------
 
+    if(!isset($_POST['date_registration_possible']))
+    {
+        $_POST['date_registration_possible'] = 0;
+    }
+    if($_POST['date_registration_possible'] == 1
+    && (!isset($_POST['adm_event_participation_right']) || array_count_values($_POST['adm_event_participation_right']) == 0))
+    {
+        $_SESSION['dates_request']['adm_event_participation_right'] = '';
```

Comment:
Until now we have no function to validate arrays. Had discuessed this with ximex before. We want to integrate a lib for validation after that we could also validate arrays. But since my last commit all methods in RolesRights will check the array for integer values. So everything with class RolesRights is save.